sympy/solvers/decompogen.py

```diff
@@ -42,7 +43,7 @@ def decompogen(f, symbol):
 
     # ===== Convert to Polynomial ===== #
     fp = Poly(f)
-    gens = fp.gens
+    gens = list(filter(lambda x: symbol in x.atoms(Symbol) , fp.gens))
```

Comment:
`filter` already returns a tuple here, you don't need to typecast it into a list.